packages/react-reconciler/src/ReactFiber.js

```diff
@@ -163,6 +164,9 @@ export type Fiber = {|
   // A linked-list of contexts that this fiber depends on
   contextDependencies: ContextDependencyList | null,
 
+  // An array of event component instances for this fiber
+  events: Array<ReactEventComponentInstance> | null,
```

Comment:
Could you put this array on the `ContextDependencyList` object instead? That way we can avoid another `fiber` field.

And then we can rename `contextDependencies` to just `dependencies`, perhaps.

Reply:
Isn't `ContextDependencyList` a linked list though. I opted for an array here, which would be a type-mismatch. Also, what if a component uses both context and events?

Reply:
There's a wrapper list object where you could attach the array. So they wouldn't be part of the same list:

```js
type Dependencies = {
  firstContext: ContextDependency,
  events: Array<ReactEventComponentInstance>
}
```